vpr/src/device/rr_graph_builder.cpp

```diff
@@ -23,6 +16,22 @@ RRSpatialLookup& RRGraphBuilder::node_lookup() {
     return node_lookup_;
 }
 
+MetadataStorage<int>& RRGraphBuilder::rr_node_metadata() {
+    return rr_node_metadata_;
+}
+
+MetadataStorage<std::tuple<int, int, short>>& RRGraphBuilder::rr_edge_metadata() {
+    return rr_edge_metadata_;
+}
+
+MetadataStorage<int> RRGraphBuilder::rr_node_metadata_data() const {
```

Comment:
In principle, 
- ``RRGraphBuilder`` should return reference/pointers which are mutable objects.
- ``RRGraphView`` should return objects (read-only).

Therefore, the API in LINE 27 and LINE 31 should be moved to ``RRGraphView``

Reply:
The two API's are moved to RRGraphView.